### PR TITLE
Optimizing Prow Presubmit

### PR DESCRIPTION
**Pull request by chxchx**
[Design Doc](https://docs.google.com/document/d/1m0cJ5yA8iFLpRcOf4IePqcO72ENIt9B0uthQKWdZjjc/edit#)

This update will be done in three step:
* Create spearate unit test and integration test Prow scripts, but also keep their commands in `istio-presubmit.sh`
* Update prow config to set up these jobs, mark them as required, and update merge-bot to also gate on these tests
* Remove these jobs from `istio-presubmit.sh`

### CODE REVIEW

File: prow/istio-presubmit.sh [hunk 1]
```diff
@@ -15,7 +15,6 @@
 # limitations under the License.
```
Comment (sebastienvas) [inline prow/istio-presubmit.sh]: Can you add lib in this file name, such that we know it is does not do anything by itself.

File: prow/istio-presubmit.sh [hunk 2]
```diff
@@ -35,42 +34,44 @@ die () {
...
+    # but we depend on being at path $GOPATH/src/istio.io/istio for imports
+    mv ${GOPATH}/src/github.com/istio ${GOPATH}/src/istio.io
+    ROOT=${GOPATH}/src/istio.io/istio
+    cd ${GOPATH}/src/istio.io/istio
+
+    # Use the provided pull head sha, from prow.
+    export GIT_SHA="${PULL_PULL_SHA}"
+
+    # check if rewrite history is present
+    PR_BRANCH=$(git show-ref | grep refs/pr | awk '{print $2}')
```
Comment (sebastienvas) [inline prow/istio-presubmit.sh]: I think it is safe to remove this now.

**Comment by yutongz** [comment]
I would suggest you to add the doc link in the PR description :)

**Comment by chxchx** [comment]
PTAL

**Comment by chxchx** [comment]
PTAL @sebastienvas

File: prow/lib.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    mv ${GOPATH}/src/github.com/istio ${GOPATH}/src/istio.io
+    ROOT=${GOPATH}/src/istio.io/istio
+    cd ${GOPATH}/src/istio.io/istio
+
+    # Use the provided pull head sha, from prow.
+    export GIT_SHA="${PULL_PULL_SHA}"
+
+    # Use volume mount from pilot-presubmit job's pod spec.
+    # FIXME pilot should not need this
+    ln -sf "${HOME}/.kube/config" pilot/pkg/kube/config
```
Comment (rkpagadala) [inline prow/lib.sh]: KUBECONFIG shell variable is being used, change to

# Use volume mount from pilot-presubmit job's pod spec.
 export KUBECONFIG="${HOME}/.kube/config"

Comment (chxchx) [reply prow/lib.sh]: Done

Comment (chxchx) [reply prow/lib.sh]: the latest commit containing this change failed prow/istio-pilot-e2e.sh
The current pre/post submit on master also does not export KUBECONFIG. Should I revert?

File: prow/istio-unit-tests.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+# No unset vars, print commands as they're executed, and exit on any non-zero
+# return code
+set -u
+set -x
+set -e
+
+source ${ROOT}/prow/lib.sh
+setup_and_export_git_sha
+
+echo 'Build'
```
Comment (sebastienvas) [inline prow/istio-unit-tests.sh]: Do you need to do the build ?

**Comment by sebastienvas** [comment]
What about the security tests ? do we need another file for them as well ?

**Comment by chxchx** [comment]
As Yutong mentioned, the security tests kind of need to be run in the same job as building artifacts to propagate the right certificates. Thus, I end up not separating them.

File: prow/istio-presubmit.sh [hunk 5]
```diff
@@ -30,47 +29,9 @@ set -u
...
-  # but we depend on being at path $GOPATH/src/istio.io/istio for imports
-  mv ${GOPATH}/src/github.com/istio ${GOPATH}/src/istio.io
-  ROOT=${GOPATH}/src/istio.io/istio
-  cd ${GOPATH}/src/istio.io/istio
-
-  # Use the provided pull head sha, from prow.
-  GIT_SHA="${PULL_PULL_SHA}"
-
-  # check if rewrite history is present
-  PR_BRANCH=$(git show-ref | grep refs/pr | awk '{print $2}')
```
Comment (yutongz) [inline prow/istio-presubmit.sh]: You need to keep it. This part is to check if the branch is cloned from old history (before mono-repo change)

Comment (sebastienvas) [reply prow/istio-presubmit.sh]: that s ok, we are well passed the monorepo change.

**Pull request merged**